File: frontend/external/src/types/Partner.ts
import type { BookingStatus } from "./Booking";

export interface Partner {
  id: number;
  ownerFirstName: string;
  ownerLastName: string;
  createdOn: string;
  contractFileName?: string | null;
  ownerIdentityFileName: string;
  registrationDate: string;
  partnershipEndDate: string;
  relatedUserId: string;
  phoneNumber: string;
}

export interface PartnerWallet {
  partnerUserId: string;
  currency: string;
  pendingAmount: number;
  availableAmount: number;
  reservedAmount: number;
  updatedAt: string;
}

export interface PartnerLedgerEntry {
  id: number;
  partnerUserId: string;
  bookingId?: number | null;
  customerPaymentId?: number | null;
  partnerPayoutId?: number | null;
  entryType: string;
  bucket: string;
  amountDelta: number;
  currency: string;
  description?: string | null;
  createdAt: string;
}

export interface PartnerBooking {
  id: number;
  userId: string;
  partnerCarId: number;
  partnerUserId: string;
  carBrand: string;
  carModel: string;
  startTime: string;
  endTime: string;
  priceHour?: number | null;
  totalPrice?: number | null;
  createdAt: string;
  status: BookingStatus;
}

File: backend/internal/partner-service/src/PartnerService.Domain/Entities/Partner.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace PartnerService.Domain.Entities;

public class Partner
{
    [Column("id")]
    public int Id { get; set; }

    [Column("owner_first_name")]
    public string OwnerFirstName { get; set; } = string.Empty;

    [Column("owner_last_name")]
    public string OwnerLastName { get; set; } = string.Empty;

    [Column("created_on")]
    public DateTime CreatedOn { get; set; }

    [Column("contract_file_name")]
    public string? ContractFileName { get; set; }

    [Column("owner_identity_file_name")]
    public string OwnerIdentityFileName { get; set; } = string.Empty;

    [Column("registration_date")]
    public DateOnly RegistrationDate { get; set; }

    [Column("partnership_end_date")]
    public DateOnly PartnershipEndDate { get; set; }

    [Column("related_user_id")]
    public string RelatedUserId { get; set; } = string.Empty;

    [Column("phone_number")]
    public string PhoneNumber { get; set; } = string.Empty;

    [Column("provision_request_key")]
    public string? ProvisionRequestKey { get; set; }
}
